src/java/org/sogive/server/CharityServlet.java

```diff
@@ -81,6 +83,11 @@ protected void doSave(WebRequest state) {
 		if (redirect == null) {
 			return thing;
 		}
+		// do not redirect in editor
+		String testGet = state.getRequest().getQueryString();
+		if (testGet.contains("noRedirect=true")) {
```

Comment:
If this works in testing then there's no need to change it.

For future - you could write: 
`testGet = state.get("noRedirect")`

Which handles both get (i.e. url) and post (i.e. request body) parameters, plus it handles decoding the url formatted string - which doesn't matter here, but you'd want for complex strings.